api/streams.go

```diff
@@ -432,6 +432,72 @@ func (p RetentionPolicy) MarshalYAML() (any, error) {
 	}
 }
 
+type Compression int
+
+const (
+	NoCompression Compression = iota
+	S2Compression
+)
+
+func (p Compression) String() string {
+	switch p {
+	case NoCompression:
+		return "None"
+	case S2Compression:
+		return "S2 Compression"
+	default:
+		return "Unknown Compression"
+	}
+}
+
+func (p *Compression) UnmarshalJSON(data []byte) error {
+	switch string(data) {
+	case jsonString("none"), jsonString(""):
+		*p = NoCompression
+	case jsonString("s2"):
+		*p = S2Compression
+	default:
+		return fmt.Errorf("can not unmarshal %q", data)
+	}
+
+	return nil
+}
+
+func (p *Compression) UnmarshalYAML(data *yaml.Node) error {
+	switch data.Value {
+	case "none", "":
```

Comment:
Out of curiosity, if we don't accept `""` as a sane default, what happens if someone tries to create a stream using an out-of-date NATS CLI?

Reply:
hmm, ugh.  Thats a good question.

I think what we did is the server handle that version upgrade to `none` so that new cli/jsm/terraform always get consistent updated content.  But its hard to say exactly without playing

Reply:
Other options is new nats cli on old server - it would not return value and so be unusable.  So I guess we have no choice but to do what you have here.